tests/php/Storage/AssetStoreTest.php

```diff
@@ -899,6 +921,98 @@ public function listOfFilesToNormalise()
         $public = AssetStore::VISIBILITY_PUBLIC;
         $protected = AssetStore::VISIBILITY_PROTECTED;
 
+        /** @var FileIDHelper $hashHelper */
+        $hashHelper = new HashFileIDHelper();
```

Comment:
For a couple of the unit-test + provider combindations, I've duplicated the methods and retained the deprecated code in methods that have `WithLegacy()` suffixed to them.